### PR TITLE
Add Chat feature

### PR DESCRIPTION
**Pull request by fbernaly**
Added Chat feature as explained here: https://heroiclabs.com/docs/nakama/client-libraries/unity/#chat
Added methods:
- joinChannel
- sendMessage
- listChannelMessages
- updateMessage
and added a StreamController to listen to incoming messages.

Moved `nakama_client_stub.dart` to a subfolder with implementations for io and js, to allow exposing `NakamaRestApiClient` and `NakamaGrpcClient` if devs prefer to instantiate one of them directly without forcing them to use `getNakamaClient`.

### CODE REVIEW

**Comment by obrunsmann** [comment]
@fbernaly Please add test cases for chat feature

**Comment by fbernaly** [comment]
@obrunsmann : I added unit tests, any update?

**Comment by obrunsmann** [comment]
@fbernaly sorry for my late reply, had been quite busy last days and some issues with my local setup 😬 I have some comments, but for now no blockings. So I'd finish my review now and then I just change the last small things on the develop branch.

File: lib/src/nakama_websocket_client.dart [hunk 1]
```diff
@@ -297,4 +326,40 @@ class NakamaWebsocketClient {
...
+          channelJoin: rtpb.ChannelJoin(
+        target: target,
+        type: type.value,
+        persistence: BoolValue(value: persistence),
+        hidden: BoolValue(value: hidden),
+      )));
+
+  Future<rtpb.ChannelMessageAck> sendMessage({
+    required String channelId,
+    required Map<String, String> content,
```
Comment (obrunsmann) [inline lib/src/nakama_websocket_client.dart:346]: Should be `<String, dynamic>` imo, to also accept int values e.g.

File: lib/src/nakama_websocket_client.dart [hunk 2]
```diff
@@ -297,4 +326,40 @@ class NakamaWebsocketClient {
...
+      _send<rtpb.ChannelMessageAck>(rtpb.Envelope(
+          channelMessageSend: rtpb.ChannelMessageSend(
+        channelId: channelId,
+        content: jsonEncode(content),
+      )));
+
+  Future<rtpb.ChannelMessageAck> updateMessage({
+    required String channelId,
+    required String messageId,
+    required Map<String, String> content,
```
Comment (obrunsmann) [inline lib/src/nakama_websocket_client.dart:357]: Same here

**Comment by fbernaly** [comment]
@obrunsmann : are you going to fix those comments in your branch?

**Pull request merged**